### ISSUE TITLE
Canonicalize cannot sort an array of hashes

### ISSUE THREAD

**Issue by Hvid** [puppetlabs/ruby-pwsh#117]
## Describe the Bug

The dsc_xwebsite resource in xwebadministration can take a parameter for bindinginfo. This parameter is supposed to be an array of structs. I'm getting this error with the puppet agent:
`Error: Failed to apply catalog: comparison of Hash with Hash failed`

This line in dsc_base_provider.rb is where it's failing:
`is_same = value.is_a?(Enumerable) & !downcased_resource[key].nil? ? downcased_resource[key].sort == value.sort : downcased_resource[key] == value`
The dsc_bindinginfo is supposed to be an array. This array contains a hash for each binding:
`:dsc_bindinginfo=>[{"protocol"=>"http", "port"=>81}, {"protocol"=>"http", "port"=>80}],`
According to the refence, it should be a struct, but i'm not sure that actually matters. The error seems to be the .sort:
`ArgumentError: comparison of Hash with Hash failed`

## Expected Behavior
I guess it needs to know how to sort the hashes if needed? 

## Steps to Reproduce
Test code:
```
dsc_xwebsite { 'splash':
    dsc_name               => 'splash',
    dsc_physicalpath       => "D:\\Splash",
    dsc_logflags           => [
      'Date',
      'Time',
      'ClientIP',
      'UserName',
      'SiteName',
      'ComputerName',
      'ServerIP',
      'Method',
      'UriStem',
      'UriQuery',
      'HttpStatus',
      'Win32Status',
      'BytesSent',
      'BytesRecv',
      'TimeTaken',
      'ServerPort',
      'UserAgent',
      'Referer',
      'ProtocolVersion',
      'Host',
      'HttpSubStatus',
    ],
    dsc_serverautostart    => true,
    dsc_state              => 'Started',
    dsc_applicationpool    => 'Splash',
    dsc_enabledprotocols   => 'http',
    dsc_defaultpage        => [
      'default.asp',
    ],
    dsc_bindinginfo        => [
      {
        protocol => 'http',
        port     => 81,
      },
      {
        protocol => 'http',
        port     => 80,
      },
    ],
    dsc_ensure             => 'Present',
    dsc_logperiod          => 'Daily',
    dsc_logpath            => "D:\\Logfiles",
    dsc_authenticationinfo => {
      anonymous => true,
      digest    => false,
      basic     => false,
      windows   => false
    },
```

## Environment
Windows - PDK 2.0.0.

## Additional Context
Slack link: https://puppetcommunity.slack.com/archives/C0W1Z7213/p1612780438062100

**Comment by Hvid** [comment]
https://github.com/puppetlabs/ruby-pwsh/pull/118 fixes this.

**Comment by michaeltlombardi** [comment]
Closed by #118!
